### PR TITLE
Feature/con 2195/info overlay

### PR DESCRIPTION
**Pull request by MWest2020**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes/
| BC breaks?    | 
| Deprecations? | 
| Tests pass?   | 
| Fixed tickets | #2195
| License       | MIT
| Doc PR        | 

Implemented the requested modal for all components with a help button

### CODE REVIEW

File: pwa/src/components/applications/applicationsTable.tsx [hunk 1]
```diff
@@ -74,7 +100,7 @@ export default function ApplicationsTable() {
                       headerName: " ",
                       renderCell: (item: { id: string }) => {
                         return (
-                          <Link className="utrecht-link d-flex justify-content-end" to={`/applications/${item.id}`}>
+                          <Link to={`/applications/${item.id}`}>
```
Comment (lencodes) [inline pwa/src/components/applications/applicationsTable.tsx:103]: Why is this change necessary? 

There's at least one thing breaking due to this change, the _Edit_ button was aligned right in its table with the classes `d-flex justify-content-end`.

File: pwa/src/components/configurations_export_button/ConfigurationsExportButton.tsx [hunk 2]
```diff
@@ -39,7 +41,7 @@ export default function ConfigurationsExportButton() {
       onClick={handleExport}
       disabled={loading}
     >
-      {loading ? "Perparing your download...": "Export Configuration"}
+      {loading ? "Preparing your download..." : "Export Configuration"}
```
Comment (lencodes) [inline pwa/src/components/configurations_export_button/ConfigurationsExportButton.tsx:44]: Nice 😄

File: pwa/src/components/endpoints/endpointsTable.tsx [hunk 3]
```diff
@@ -31,105 +41,139 @@ export default function EndpointsTable() {
...
-              <i className="fas fa-sync-alt mr-1" />
-              <span className="mr-2">Refresh</span>
-            </a>
-            <Link to="/endpoints/new">
-              <button className="utrecht-button utrecht-button-sm btn-sm btn-success">
-                <i className="fas fa-plus mr-2" />
-                Create
+  React.useEffect(() => {
+    // na apiService refactor
+    // handleSetApplications();
```
Comment (lencodes) [inline pwa/src/components/endpoints/endpointsTable.tsx:61]: Remove comments.

File: pwa/src/components/entities/entitiesTable.tsx [hunk 4]
```diff
@@ -1,29 +1,48 @@
...
-    setShowSpinner(true)
+    setShowSpinner(true);
     API.Entity.getAll()
       .then((res) => {
-        setEntities(res.data)
+        setEntities(res.data);
       })
       .catch((err) => {
-        throw new Error ('GET Applications error: ' + err)
+        throw new Error("GET Applications error: " + err);
```
Comment (lencodes) [inline pwa/src/components/entities/entitiesTable.tsx:30]: GET Entities error*

File: pwa/src/components/sources/sourcesTable.tsx [hunk 5]
```diff
@@ -1,36 +1,67 @@
...
-    setShowSpinner(true)
+    setShowSpinner(true);
     API.Source.getAll()
       .then((res) => {
-        setSources(res.data)
+        setSources(res.data);
       })
       .catch((err) => {
-        throw new Error ('GET Applications error: ' + err)
+        throw new Error("GET Applications error: " + err);
```
Comment (lencodes) [inline pwa/src/components/sources/sourcesTable.tsx:30]: Get Sources error*

File: pwa/src/styles/main.css [hunk 6]
```diff
@@ -219,3 +219,8 @@ a:hover {
     border: none !important;
     background: inherit !important;
 }
+
+#helpModal {
+    text-align: left;
+    color: black;
+}
```
Comment (lencodes) [inline pwa/src/styles/main.css:226]: Usually we reserve using an `id` (`#`-selector) for edge case javascript selecting (it was very popular in things like `jQuery` and is still in use in things like `Bootstrap` so that's why we need to actually add the `#helpModal` id at this moment).

In React however, we pretty much never want to add an `id` to a component. Therefore, we also would like to not use the `#` selector in our `css`.

Above being said, usually we would have a unique `className` for each component and we can scope our styling like that. However, because of the way everything's structured right now, that would be a bit of a rework.

Therefore: I would like to suggest to just swap this `#helpModal` with `.modal` 👍🏼

**Review by lencodes — CHANGES_REQUESTED** [review]
Looks good! Couple of small comments.

**Review by MWest2020 — COMMENTED** [review]
Please review changes so I don't merge something and break dev branch
